### ISSUE TITLE
Extra call node creation

### ISSUE THREAD

**Issue by EJBQ** [prmr/JetUML#73]
This is again with sequence diagrams.

Steps to reproduce:
Create two implicit parameter nodes
Give them one call child each
Connect the call nodes
Delete the target call node
Undo the deletion

What this does, which you can check by giving the edge and node non-default properties, is create a new call node at the end of the same edge when it is put back. Then the original call node is placed below the new one. Both new call nodes are children of the original parent call node.

If you delete the commits one by one going into the past, I'm seeing this issue appear with the merge of edge lists in nodes.

**Comment by prmr** [comment]
Weird, in my case the undo doesn't even work.

**Comment by EJBQ** [comment]
Quick video of the issue as I'm seeing it: https://vid.me/S9iZ

**Comment by prmr** [comment]
Ah, I get it. Is the callee still a child node? I thought we had refactored this away.

**Comment by EJBQ** [comment]
The callee is still a listed as part of aCalls, which is the separate storage that calls have for their children now.

So what I'm seeing with this is the edge is never added to originEdges or terminalEdges for either callNode. Then when the edge is deleted, because it is not in those lists, when removeEdge is called on the nodes, they return true, which breaks the for loop of notifying the edges.. I'm not sure in what cases that method is supposed to return true or false though.

I just pushed a change that I think fixes it, but there definitely still issues with the origin and terminal edge lists that I pointed to above.
